api/src/test/java/org/apache/iceberg/types/TestTypeUtil.java

```diff
@@ -85,33 +89,33 @@ public void testAssignIncreasingFreshIdWithIdentifier() {
         .isEqualTo(expectedSchema.identifierFieldIds());
   }
 
-  @Test
-  public void testAssignIncreasingFreshIdNewIdentifier() {
+  @ParameterizedTest
+  @MethodSource("testTypes")
+  public void testAssignIncreasingFreshIdNewIdentifier(Type testType) {
```

Comment:
Please roll back changes to this test.

Reply:
@aihuaxu, it looks like I'd be pasting this message a lot. Please roll back all changes to this test suite and add new behavior tests to exercise the relevant behavior for variants.